### ISSUE TITLE
Problems with src/makefile.in for 4.7.0

### ISSUE THREAD

**Issue by Rasmus-C** [vrogier/ocilib#243]
Hi,

I tried to update our ocilib to the new release 4.7.0. but struggled with problems like:
**gmake[2]: *** No rule to make target 'string.c', needed by 'libocilib_la-string.lo'.  Stop.**

_Using gcc 7.2.0 and rhel6_

Building 4.6.4 works fine and there the file src/string.c existed. If I see it correctly, it has been renamed to src/strings.c in 4.7.0. I thus checked why this is used and found places in src/Makefile.in where I replaced string.c by strings.c. Furthermore, I got several other messages and made some more adjustments to the Makefile.in . With these changes I was able to compile and link it. However, not sure if any other place needs to be modified as well and, of course, I wanted to report the issue.

Best regards and thanks for continuously improving the ocilib.

```
diff --git a/src/Makefile.in b/src/Makefile.in
index f6c8453..ddc8d7c 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -425,8 +425,10 @@ include_HEADERS=                             \
     $(top_srcdir)/include/ocilibc/config.h   \
     $(top_srcdir)/include/ocilibc/types.h    \
     $(top_srcdir)/include/ocilib.hpp         \
+    $(top_srcdir)/include/ocilibcpp/config.hpp \
     $(top_srcdir)/include/ocilibcpp/core.hpp \
-    $(top_srcdir)/include/ocilibcpp/impl.hpp
+    $(top_srcdir)/include/ocilibcpp/support.hpp \
+    $(top_srcdir)/include/ocilibcpp/types.hpp
 noinst_HEADERS=     \
 	agent.h         \
     array.h         \
@@ -703,12 +705,12 @@ libocilib_la-resultset.lo: resultset.c
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libocilib_la_CFLAGS) $(CFLAGS) -c -o libocilib_la-resultset.lo `test -f 'resultset.c' || echo '$(srcdir)/'`resultset.c
 
-libocilib_la-string.lo: string.c
-@am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libocilib_la_CFLAGS) $(CFLAGS) -MT libocilib_la-string.lo -MD -MP -MF $(DEPDIR)/libocilib_la-string.Tpo -c -o libocilib_la-string.lo `test -f 'string.c' || echo '$(srcdir)/'`string.c
+libocilib_la-string.lo: strings.c
+@am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libocilib_la_CFLAGS) $(CFLAGS) -MT libocilib_la-string.lo -MD -MP -MF $(DEPDIR)/libocilib_la-string.Tpo -c -o libocilib_la-string.lo `test -f 'strings.c' || echo '$(srcdir)/'`strings.c
 @am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) $(DEPDIR)/libocilib_la-string.Tpo $(DEPDIR)/libocilib_la-string.Plo
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='string.c' object='libocilib_la-string.lo' libtool=yes @AMDEPBACKSLASH@
+@AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='strings.c' object='libocilib_la-string.lo' libtool=yes @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libocilib_la_CFLAGS) $(CFLAGS) -c -o libocilib_la-string.lo `test -f 'string.c' || echo '$(srcdir)/'`string.c
+@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libocilib_la_CFLAGS) $(CFLAGS) -c -o libocilib_la-string.lo `test -f 'strings.c' || echo '$(srcdir)/'`strings.c
 
 libocilib_la-timestamp.lo: timestamp.c
 @am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libocilib_la_CFLAGS) $(CFLAGS) -MT libocilib_la-timestamp.lo -MD -MP -MF $(DEPDIR)/libocilib_la-timestamp.Tpo -c -o libocilib_la-timestamp.lo `test -f 'timestamp.c' || echo '$(srcdir)/'`timestamp.c
@@ -752,12 +754,12 @@ libocilib_la-hash.lo: hash.c
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libocilib_la_CFLAGS) $(CFLAGS) -c -o libocilib_la-hash.lo `test -f 'hash.c' || echo '$(srcdir)/'`hash.c
 
-libocilib_la-library.lo: library.c
-@am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libocilib_la_CFLAGS) $(CFLAGS) -MT libocilib_la-library.lo -MD -MP -MF $(DEPDIR)/libocilib_la-library.Tpo -c -o libocilib_la-library.lo `test -f 'library.c' || echo '$(srcdir)/'`library.c
+libocilib_la-library.lo: ocilib.c
+@am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libocilib_la_CFLAGS) $(CFLAGS) -MT libocilib_la-library.lo -MD -MP -MF $(DEPDIR)/libocilib_la-library.Tpo -c -o libocilib_la-library.lo `test -f 'ocilib.c' || echo '$(srcdir)/'`ocilib.c
 @am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) $(DEPDIR)/libocilib_la-library.Tpo $(DEPDIR)/libocilib_la-library.Plo
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='library.c' object='libocilib_la-library.lo' libtool=yes @AMDEPBACKSLASH@
+@AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='ocilib.c' object='libocilib_la-library.lo' libtool=yes @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libocilib_la_CFLAGS) $(CFLAGS) -c -o libocilib_la-library.lo `test -f 'library.c' || echo '$(srcdir)/'`library.c
+@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libocilib_la_CFLAGS) $(CFLAGS) -c -o libocilib_la-library.lo `test -f 'ocilib.c' || echo '$(srcdir)/'`ocilib.c
 
 libocilib_la-long.lo: long.c
 @am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libocilib_la_CFLAGS) $(CFLAGS) -MT libocilib_la-long.lo -MD -MP -MF $(DEPDIR)/libocilib_la-long.Tpo -c -o libocilib_la-long.lo `test -f 'long.c' || echo '$(srcdir)/'`long.c
```

**Comment by vrogier** [comment]
Hi,

Thanks for reporting this issue.
You must try to compile directly from GIT repository instead of using the distribution archive that contains correct configure scripts and makefiles
Some scripts were not indeed updated into source control.
I've fixed it in v4.7.1 branch (release scheduled next week).

Best regards,

Vincent

**Comment by vrogier** [comment]
fixed by commit 391dae4a

**Comment by vrogier** [comment]
Branch develop-v4.7.1 merged into master
